packages/contracts/test/hashchain.test.ts

```diff
@@ -132,7 +132,11 @@ describe('Hashchain tests', function () {
             .connect(attester)
             .submitAttestation(epoch, epochKey, 1, 0, 0)
             .then((t) => t.wait())
-        const hashchainIndex = 0
+        const hashchainIndex =
```

Comment:
We need to check this value, it should always be 0.
 
It could be like:
```
const hashchainIndex = await unirepContract.attesterHashchainProcessedCount(
  attester.address,
  epoch
)
expect(hashchainIndex).to.equal(0)
```

But this roughly simplifies to `const hashchainIndex = 0`

Reply:
Maybe it's good to explicitly load and check. If we hardcode 0 It's possible we make multiple chains and they are the same, then it false passes.

Reply:
I think we should add another test for `attesterHashchainProcessedCount` which computes the right `hashchainIndex`.
but here I prefer to load it from contract (if we are not testing for that)
I am thinking that if we want to copy-paste this test code to other place it would be better to show how to get this data.
like if we want to test other thing using `genEpochKey`
if would be hard to know from `genEpochKey(idNullifier, 0, 0)` where one `0` is `epoch` and it can be queried from contract

Reply:
> I think we should add another test for attesterHashchainProcessedCount which computes the right hashchainIndex

Yes i agree

> I am thinking that if we want to copy-paste this test code to other place it would be better to show how to get this data.

Ehh, i don't think we should write code to optimize for copy/paste-ability. But it's fine to load from contract as long as we have an assertion like

```ts
expect(hashchainIndex).to.equal(0)
```